dev_apps/music-nga/elements/music-artwork.js

```diff
@@ -187,12 +199,28 @@ proto.createdCallback = function() {
     }));
   });
 
+  this.onDOMLocalized = () => {
+    // XXX: Bug 1205799 - view.formatValue errors when called before first
+    // language is resolved
+    document.l10n.ready.then(() => {
+      document.l10n.translateFragment(shadowRoot);
+    });
+  };
+
   this.repeat = this.getAttribute('repeat');
   this.shuffle = this.getAttribute('shuffle');
 
   this.overlayVisible = true;
 };
 
+proto.attachedCallback = function() {
+  document.addEventListener('DOMLocalized', this.onDOMLocalized);
+};
+
+proto.detachedCallback = function() {
+  document.removeEventListener('DOMLocalized', this.onDOMLocalized);
```

Comment:
Here, you don't need to guard `translateFragment` with a `ready` because the `DOMLocalized` event guarantees that the localization is ready.  This should work:

    const onDOMLocalized = () => document.l10n.translateFragment(shadowRoot);
    document.addEventListener('DOMLocalized', onDOMLocalized);
    document.removeEventListener('DOMLocalized', onDOMLocalized);